src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +152,63 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            if (_encoding == null || _decoder == null)
+            {
+                _encoding = Encoding.UTF8;
+                _decoder = _encoding.GetDecoder();
+            }
+
+            char[]? chars = null;
+            byte[]? bytes = null;
+            const int size = 4096;
             try
             {
-                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                using var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false);
+
+                bytes = ArrayPool<byte>.Shared.Rent(size);
+                chars = ArrayPool<char>.Shared.Rent(_encoding.GetMaxCharCount(size));
+                int charCount = 0;
+
+                while (true)
                 {
-                    fileContents = reusableReader.ReadAllText(fileStream);
-                    return true;
+                    int count = size;
+                    int index = 0;
+                    while (count > 0)
+                    {
+                        int read = fileStream.Read(bytes, index, count);
+                        if (read == 0)
+                        {
+                            contents = new string(chars, 0, charCount);
+                            return true;
+                        }
+
+                        charCount += _decoder.GetChars(bytes, 0, read, chars, 0);
```

Comment:
_decoder is stateful.  Using a shared instance like this from potentially multiple threads concurrently isn't safe.